gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -123,6 +123,89 @@ def test_{{ method_name }}_empty_call():
         {% endif %}
 {% endif %}
 
+def test_{{ method_name }}_use_cached_wrapped_rpc():
+    # Clients should use _prep_wrapped_messages to create cached wrapped rpcs,
+    # instead of constructing them on each call
+    with mock.patch("google.api_core.gapic_v1.method.wrap_method") as wrapper_fn:
+        client = {{ service.client_name }}(
+            credentials=ga_credentials.AnonymousCredentials(),
+            transport="grpc",
+        )
+
+        # Should wrap all calls on client creation
+        assert wrapper_fn.call_count > 0
+        wrapper_fn.reset_mock()
+
+        # Ensure method has been cached
+        assert client._transport.{{method.transport_safe_name|snake_case}} in client._transport._wrapped_methods
+
+        # Replace cached wrapped function with mock
+        mock_rpc = mock.Mock()
+        client._transport._wrapped_methods[client._transport.{{method.transport_safe_name|snake_case}}] = mock_rpc
+
+        request = {}
+        {% if method.client_streaming %}
+        requests = [request]
+        client.{{ method.safe_name|snake_case }}(iter(requests))
+        {% else %}
+        client.{{ method_name }}(request)
+        {% endif %}
+
+        # Establish that the underlying gRPC stub method was called.
+        assert mock_rpc.call_count == 1
+
+        # Operation methods build a cached wrapper on first rpc call
```

Comment:
Could we make this stanza conditional on the method in fact being an Operation method?